diff --git a/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java b/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
index 1c12d1fd5ef3d2580474893b49a1d27e18ae0923..3eb1f156c3e506816a911df7edd9504c21351b31 100644
--- a/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
+++ b/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
@@ -76,3 +76,2 @@ public class RNWidget {
         Bitmap bitmap = drawViewToBitmap(widgetWithViews.getRootView());
-        Uri bitmapUri = saveBitmapToDisk(widgetId, "light", bitmap);
-        remoteWidgetView.setImageViewUri(R.id.rn_widget_image_light, bitmapUri);
+        remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_light, bitmap);
@@ -92,3 +91,2 @@ public class RNWidget {
             Bitmap darkBitmap = drawViewToBitmap(darkWidgetWithViews.getRootView());
-            Uri darkBitmapUri = saveBitmapToDisk(widgetId, "dark", darkBitmap);
-            remoteWidgetView.setImageViewUri(R.id.rn_widget_image_dark, darkBitmapUri);
+            remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_dark, darkBitmap);
@@ -104 +102 @@ public class RNWidget {
-            remoteWidgetView.setImageViewUri(R.id.rn_widget_image_dark, bitmapUri);
+            remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_dark, bitmap);
@@ -146,6 +143,0 @@ public class RNWidget {
-    private Uri saveBitmapToDisk(int widgetId, String mode, Bitmap bitmap) {
-        String fileName = "widget_" + widgetId + "_mode_" + mode + ".png";
-        RNWidgetImageProvider.writeImage(appContext, fileName, bitmap);
-        return RNWidgetImageProvider.getImageUri(appContext, fileName);
-    }
-
